packages/codegen-ui-react/lib/amplify-ui-renderers/collection.ts

```diff
@@ -16,47 +16,184 @@
+  StudioComponent,
   StudioComponentChild,
+  StudioNode,
 } from '@aws-amplify/codegen-ui';
-import { factory, JsxChild, JsxElement, JsxExpression, JsxOpeningElement, SyntaxKind } from 'typescript';
+import ts, {
+  factory,
+  JsxAttribute,
+  JsxChild,
+  JsxElement,
+  JsxExpression,
+  JsxOpeningElement,
+  SyntaxKind,
+} from 'typescript';
 import { ReactComponentRenderer } from '../react-component-renderer';
 import { buildOpeningElementProperties } from '../react-component-render-helper';
+import { ImportCollection, ImportValue } from '../imports';
+import { ReactRenderConfig } from '../react-render-config';
+import { defaultRenderConfig } from '../react-studio-template-renderer-helper';
 
 export default class CollectionRenderer extends ReactComponentRenderer<BaseComponentProps> {
+  constructor(
+    component: StudioComponent | StudioComponentChild,
+    protected componentMetadata: ComponentMetadata,
+    protected importCollection: ImportCollection,
+    protected renderConfig: ReactRenderConfig & typeof defaultRenderConfig,
+    protected parent?: StudioNode,
+  ) {
+    super(component, componentMetadata, importCollection, parent);
+  }
+
   renderElement(renderChildren: (children: StudioComponentChild[]) => JsxChild[]): JsxElement {
     this.addKeyPropertyToChildren(this.component.children ?? []);
     const childrenJsx = this.component.children ? renderChildren(this.component.children ?? []) : [];
 
     const arrowFuncExpr = this.renderItemArrowFunctionExpr(childrenJsx);
     const itemsVariableName = this.findItemsVariableName();
-    const element = factory.createJsxElement(
-      this.renderCollectionOpeningElement(itemsVariableName),
-      [arrowFuncExpr],
-      factory.createJsxClosingElement(factory.createIdentifier(this.component.componentType)),
-    );
+    const element =
+      this.renderConfig.apiConfiguration?.dataApi === 'GraphQL'
+        ? factory.createJsxElement(
+            factory.createJsxOpeningElement(
+              factory.createIdentifier('div'),
+              undefined,
+              factory.createJsxAttributes([]),
+            ),
+            [
+              factory.createJsxElement(
+                this.renderCollectionOpeningElement(itemsVariableName),
+                [arrowFuncExpr],
+                factory.createJsxClosingElement(factory.createIdentifier(this.component.componentType)),
+              ),
+              factory.createJsxExpression(
+                undefined,
+                factory.createBinaryExpression(
+                  factory.createIdentifier('isApiPagination'),
+                  factory.createToken(ts.SyntaxKind.AmpersandAmpersandToken),
+                  factory.createJsxSelfClosingElement(
+                    factory.createIdentifier('Pagination'),
+                    undefined,
+                    factory.createJsxAttributes([
+                      factory.createJsxAttribute(
+                        factory.createIdentifier('currentPage'),
```

Comment:
A lot of this is boilerplate. Can we refactor to just have a list of props we can iterate over?

Reply:
Yeah, I did it in other places but missed this one.